sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -553,7 +555,10 @@ private[hive] class TestHiveQueryExecution(
     val referencedTables =
       describedTables ++
         logical.collect { case UnresolvedRelation(tableIdent, _) => tableIdent.table }
-    val referencedTestTables = referencedTables.filter(sparkSession.testTables.contains)
+    val formattedRefTables = referencedTables.map { t =>
+      if (sparkSession.sessionState.conf.caseSensitiveAnalysis) t else t.toLowerCase(Locale.ROOT)
+    }
+    val referencedTestTables = formattedRefTables.filter(sparkSession.testTables.contains)
```

Comment:
It would be an `O(N^2)` operation, but since the number of referenced tables should be small, I think that's ok.